### DISCUSSION TITLE
Important questions for me

### DISCUSSION THREAD

**Discussion by ZakolkaTV** [Somfic/EliteVA#18] [Q&A] [answered]
@Somfic, I'm sorry to bother you, but I have a very important question for you.

The fact is that I am seriously engaged in writing voice control for ED. At first, I used Laitis and created a ready-made voice control for more than 200 commands. I am also creating my own voice pack, which already has more than 1000 phrases. More than 900 pilots already use my voice control.

Now, I have started using VoiceAttack and have a serious choice - EliteVA or EDDI. EliteVA attracts me more - it is a site with convenient functionality and the ability of EliteVA to read binds and press buttons in the game, and of course huge opportunities.

It may take me more than one year to create a complete voice control for ED.

And I am very concerned about the questions:
- Will you update the plugin after in-game updates, if necessary?
- Can I count on a quick fix for critical errors in the plugin that make it unusable?
- If you need to add new functionality to the plugin, are you ready to do it?

This is very important to me because I will spend a lot of time on it. And as always, I'm sorry for my English.

**Comment by Somfic** [comment, marked as answer]
Hey @ZakolkaTV,

First of all thank you for considering EliteVA for this project. EliteVA is my passion project so I'll always continue improving it if people have a use for it. However, I am currently a student, so my academic responsibilities have priority over EliteVA, meaning that sometimes I cannot get to a bug or feature request directly.

To answer your questions:

> ### Will you update the plugin after in-game updates, if necessary?

The plugin is designed to be as future-proof as possible. Meaning that new events added to the game will automatically be picked up by the plugin, without needing any updates from me. This also goes for the keybindings. 

However, this does not mean that the plugin will never need updates. There might be a bug, or some core code needs to be updated for a new game version. (This is the case with the errors you mentioned earlier in #16).

> ### Can I count on a quick fix for critical errors in the plugin that make it unusable?

For bugs that cause the EliteVA plugin to be completely unusable I'll always try to push out a fix as quickly as possible. The plugin has been around for a few years now, so luckily there's less and less critical bugs that make the plugin unusable each time. 

I am always happy to help and assist with errors. Though I must admit that I have slowly lost interest in playing Elite: Dangerous myself anymore. Meaning that I can only fix bugs that you and other users report to me. 

> ### If you need to add new functionality to the plugin, are you ready to do it?

This depends on the functionality. For example, the [live tools for the plugin](https://docs.somfic.dev/projects/eliteva/live) were requested by the community to make using the plugin easier. As mentioned before, due to academic reasons I dont have as much free time as I had before, so development for big features may be slower and I can't always guarantee them.

This is also why EliteVA is free and open-source, since I cannot make any guarantees, as my life can get really busy. Anyone with coding experience is free to contribute to the project, to help keep the plugin up-to-date and error free for the latest game versions.

**In short; EliteVA is my baby passion project, so I'll always keep working on it if people are interested. I cannot make any promises when it comes to time frames or instant updates, as this is a hobby project for me.**

I hope this answered your questions and gave you some more insight into the support plans for EliteVA.

Let me know if you have any other questions!

:)

**Comment by ZakolkaTV** [comment]
@Somfic, after such a detailed answer, I have no questions left. Thank you for staying in touch and not abandoning the project, I'm sure this is important for many pilots, and especially for me.

In the near future, I will actively use your plugin to write voice control, so if something goes wrong, I will immediately let you know =)

It is important for me that critical errors that make the plugin not workable are fixed as quickly as possible, everything else can wait as long as necessary.

Thank you so much for your invaluable work.
